.github/workflows/tests-contributors.yml

```diff
@@ -37,30 +37,77 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Generate matrix for test jobs
+        id: generate_matrix
+        run: |
+          FEDVERS=34
+          TARGET_BRANCH=${{ fromJson(steps.pr_api.outputs.data).base.ref }}
+
+          JSON_MASTER_ELN='{"release": "master", "target-branch": "master", "ci-tag": "master"}, {"release": "eln", "target-branch": "master", "ci-tag": "eln", "build-args": "--build-arg=image=quay.io/fedoraci/fedora:eln-x86_64"}'
+          JSON_DEVEL='{"release": "f'$FEDVERS'-devel", "target-branch": "f'$FEDVERS'-devel", "ci-tag": "f'$FEDVERS'-devel"}'
+          JSON_RELEASE='{"release": "f'$FEDVERS'-release", "target-branch": "f'$FEDVERS'-release", "ci-tag": "f'$FEDVERS'-release"}'
+
+          REF_DEVEL='f'$FEDVERS'-devel'
+          REF_RELEASE='f'$FEDVERS'-release'
+
+          # Always order things so that first array element is the current branch, because... wait for it...
+          if [[ "$TARGET_BRANCH" == "$REF_DEVEL" ]] ; then
+            MERGED='{"include": ['$JSON_DEVEL', '$JSON_RELEASE', '$JSON_MASTER_ELN']}' ;
+          elif [[ "$TARGET_BRANCH" == "$REF_RELEASE" ]] ; then
+            MERGED='{"include": ['$JSON_RELEASE']}' ;
+          elif [[ "$TARGET_BRANCH" == "master" ]] ; then
+            MERGED='{"include": ['$JSON_MASTER_ELN']}' ;
+          else
+            OUT_JSON_FULL=''
+            OUT_JSON_PRIMARY=''
+            OUT_JSON_SECONDARY=''
```

Comment:
I guess this is just WIP. Otherwise I don't quite understand why would this else clause be here. There is no reason to create anything if it's not devel, release or master so I would just add `exit 0`.

Also these variables are not set before.

Reply:
Hm, ok, I think this needs also testing on some other branch, but let's see.

Reply:
It works fine! Thanks!